dev/build.go

```diff
@@ -0,0 +1,83 @@
+package dev
+
+import (
+	"fmt"
+	"github.com/magefile/mage/mg"
+	"github.com/magefile/mage/sh"
+	"log"
+	"os"
+	"os/exec"
+)
+
+type ImageBuildInfo struct {
+	ImageTag      string
+	CacheDir      string
+	ContainerFile string
+	ContextDir    string
+	Runtime       string
+}
+
+type ImagePushInfo struct {
+	DigestFile string
+	BuildInfo  *ImageBuildInfo
+}
+
+func must(err error) {
+	if err != nil {
+		panic(err)
+	}
+}
+
+// BuildImage is a generic image build function,
+// requires the binaries to be built beforehand
+func BuildImage(buildInfo *ImageBuildInfo, deps []interface{}) {
+	if len(deps) > 0 {
+		mg.SerialDeps(deps...)
+	}
+
+	buildCmd := []string{buildInfo.Runtime, "build", "-t", buildInfo.ImageTag}
+	if buildInfo.ContainerFile != "" {
+		buildCmd = append(buildCmd, "-f", buildInfo.ContainerFile)
+	}
+	buildCmd = append(buildCmd, buildInfo.ContextDir)
+
+	cmds := [][]string{
+		buildCmd,
+		{buildInfo.Runtime, "image", "save", "-o", buildInfo.CacheDir + ".tar", buildInfo.ImageTag},
+	}
+
+	// Build image!
+	for _, command := range cmds {
+		buildCmd := exec.Command(command[0], command[1:]...)
```

Comment:
I would just be a little careful about overriding variable names, it could make stuff harder to debug later :)

Reply:
Renamed! ✅